Add more detail to sealed definition (#4870)
dart-lang/site-www: src/language/class-modifiers.md

The `sealed` modifier prevents a class from being extended or
implemented outside its own library.

Sealed classes are implicitly [abstract](#abstract).
They cannot be constructed themselves,

> but they can define constructors for their subclasses to use.
> Subclasses of sealed classes can be constructed, 
> and sealed classes can have
> [factory constructors](/language/constructors#factory-constructors).

Reviewer: I wonder if we can combine these statements better? It seems a little convoluted to say they can have constructors for their subclasses. Then describe subclasses. Then describe another type of constructor they can have.

Also reading this again, it's also not guaranteed that their subclasses can be constructed (since they may be abstract as well), it's just that they are not automatically given this same restriction. Maybe we can just drop the mention of "subclasses can not be constructed" or modify it a bit? Not sure.

```suggestion
but they can define constructors for their subclasses to use and have 
[factory constructors](/language/constructors#factory-constructors).
Subclasses of sealed classes are, however, not implicitly abstract.
```

Author: I took this suggestion but broke it up into  bullets, lmk if that still makes sense or if it puts weird emphasis
